modules/data-streams/src/test/java/org/elasticsearch/datastreams/lifecycle/UpdateDataStreamGlobalRetentionServiceTests.java

```diff
@@ -0,0 +1,235 @@
+            1,
+            null,
+            false,
+            DataStreamLifecycle.newBuilder().dataRetention(TimeValue.timeValueDays(365)).build(),
+            List.of()
+        );
+        builder.put(dataStreamWithLifecycleLongRetention);
+        ClusterState clusterState = ClusterState.builder(ClusterName.DEFAULT).metadata(builder).build();
+        // No global retention
+        {
+            var affectedDataStreams = service.determineAffectedDataStreams(null, clusterState);
+            assertThat(affectedDataStreams.isEmpty(), is(true));
+        }
+        // No difference in global retention
+        {
+            var globalRetention = randomNonEmptyGlobalRetention();
+            var clusterStateWithRetention = ClusterState.builder(clusterState)
+                .putCustom(DataStreamGlobalRetention.TYPE, globalRetention)
+                .build();
+            var affectedDataStreams = service.determineAffectedDataStreams(globalRetention, clusterStateWithRetention);
+            assertThat(affectedDataStreams.isEmpty(), is(true));
+        }
+        // Default retention in effect
+        {
+            var globalRetention = new DataStreamGlobalRetention(TimeValue.timeValueDays(randomIntBetween(1, 10)), null);
+            var affectedDataStreams = service.determineAffectedDataStreams(globalRetention, clusterState);
+            assertThat(affectedDataStreams.size(), is(1));
+            var dataStream = affectedDataStreams.get(0);
+            assertThat(dataStream.dataStreamName(), equalTo(dataStreamNoRetention));
+            assertThat(dataStream.previousEffectiveRetention(), nullValue());
+            assertThat(dataStream.newEffectiveRetention(), equalTo(globalRetention.getDefaultRetention()));
+        }
+        // Max retention in effect
+        {
+            var globalRetention = new DataStreamGlobalRetention(null, TimeValue.timeValueDays(randomIntBetween(10, 90)));
+            var affectedDataStreams = service.determineAffectedDataStreams(globalRetention, clusterState);
+            assertThat(affectedDataStreams.size(), is(2));
+            var dataStream = affectedDataStreams.get(0);
+            assertThat(dataStream.dataStreamName(), equalTo(dataStreamLongRetention));
+            assertThat(dataStream.previousEffectiveRetention(), notNullValue());
+            assertThat(dataStream.newEffectiveRetention(), equalTo(globalRetention.getMaxRetention()));
+            dataStream = affectedDataStreams.get(1);
+            assertThat(dataStream.dataStreamName(), equalTo(dataStreamNoRetention));
+            assertThat(dataStream.previousEffectiveRetention(), nullValue());
+            assertThat(dataStream.newEffectiveRetention(), equalTo(globalRetention.getMaxRetention()));
+        }
+    }
+
+    public void testUpdateAndAcknowledgeListener() throws Exception {
+        ActionListener<UpdateDataStreamGlobalRetentionResponse> listener = new ActionListener<>() {
```

Comment:
What is this method trying to test? Based on the method name, I thought maybe it was testing that the listener was called. But it doesn't look like the listener is ever called, right? And the second listener is identical to the first listener. We don't ever call `onResponse` or `onFailure` on either one. This might all be cleared up with a comment in the test?

Reply:
You are right, so, the goal of the test is to test that the change goes through the listener gets notified. But I cannot get the listener to get notified. I will put some more effort in this. If I do not succeed I think we can rely on the java rest test that we already have.

Reply:
Temporarily removed, I checked in other cases we also do not test if the listener is called in a unit test, so it would be nice to have, but maybe not a blocker if you agree.